components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/util/NotificationReceiversRetrievalUtil.java

```diff
@@ -216,7 +217,7 @@ private static boolean isEffectiveUserStore(RealmConfiguration realmConfiguratio
                                     (IdentityTenantUtil.getTenantId(tenantDomain)).getUserStoreManager();
 
                             if (userStoreDomain != null &&
-                                    userStoreDomain.equals(UserCoreUtil.extractDomainFromName(userName))) {
+                                    userStoreDomain.toUpperCase(Locale.ROOT).equals(UserCoreUtil.extractDomainFromName(userName))) {
```

Comment:
Can't we use `userStoreDomain.equalsIgnoreCase(UserCoreUtil.extractDomainFromName(userName)`